save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/topbar/TopBarUserField.kt

```diff
@@ -0,0 +1,126 @@
+@file:Suppress("HEADER_MISSING_IN_NON_SINGLE_CLASS_FILE")
+
+package com.saveourtool.save.frontend.components.topbar
+
+import com.saveourtool.save.domain.Role
+import com.saveourtool.save.frontend.components.modal.logoutModal
+import com.saveourtool.save.frontend.externals.fontawesome.*
+import com.saveourtool.save.v1
+import com.saveourtool.save.validation.FrontendRoutes
+
+import csstype.ClassName
+import react.FC
+import react.dom.aria.*
+import react.dom.html.ReactHTML
+import react.router.useNavigate
+import react.useEffect
+import react.useState
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.isActive
+
+val topBarUserField = topBarUserField()
+
+/**
+ * Displays the user's field
+ */
+@Suppress("MAGIC_NUMBER", "LongMethod", "TOO_LONG_FUNCTION")
+private fun topBarUserField() = FC<TopBarProps> { props ->
+    val (isLogoutModalOpen, setIsLogoutModalOpen) = useState(false)
+    val (isAriaExpanded, setIsAriaExpanded) = useState(false)
+    val scope = CoroutineScope(Dispatchers.Default)
+    val navigate = useNavigate()
+    useEffect {
+        cleanup {
+            if (scope.isActive) {
+                scope.cancel()
+            }
+        }
+    }
+    ReactHTML.ul {
```

Comment:
Maaaaaaan, please check the code before asking for review... Each and every time you have the same problem
```suggestion
    ul {
```